### DISCUSSION TITLE
Search and filtering using SplitAtom?

### DISCUSSION THREAD

**Discussion by ddeisadze** [pmndrs/jotai#2343] [Q&A]
I have a list of objects. I am using SplitAtom to create individual primative atoms from the list. 

In my container to display the list of objects, I would like to filter based on search terms and other properties and then pass down the PrimativeAtom, is there anyway to do this?

**Comment by dai-shi** [comment]
I think it's possible, regardless of `splitAtom`. Is the search term in an atom?

**Reply by 0livare** [reply]
@dai-shi (I have the same question) and yes, the search term is in the atom. 

```js
const dataAtom = atom([{page: 1}, {page: 1}, {page: 2}])
const dataAtomsAtom = splitAtom(dataAtom) // creates array of 3 atoms with individual objects
```

Now I want to filter `dataAtomsAtom` based on some condition. 

With the original array (not in an atom) for example I could do something like `data.filter(x => x.page === 1)`, but how to I efficiently accomplish the same when each object is contained within an atom?

**Reply by dai-shi** [reply]
```js
atom((get) => get(dataAtomsAtom).filter((dataAtom) => get(dataAtom).page === 1))
```

I'm not sure if it's the most efficient. It's probably the same as to filter before splitting.
